bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -481,17 +456,17 @@ private static boolean isConvertibleToRecordType(Object sourceValue, BRecordType
 
             if (isFromJson) {
                 if (targetFieldTypes.containsKey(fieldName)) {
-                    if (getConvertibleTypesFromJson(valueEntry.getValue(), targetFieldTypes.get(fieldName),
-                            fieldNameLong, unresolvedValues, errors).size() != 1) {
+                    if (getConvertibleTypeFromJson(valueEntry.getValue(), targetFieldTypes.get(fieldName),
+                            fieldNameLong, unresolvedValues, errors, allowNumericConversion) ==  null) {
                         addErrorMessage(errors.size() - initialErrorCount, errors, "field '" +
                                 fieldNameLong + "' in record '" + targetType + "' should be of type '" +
                                 targetFieldTypes.get(fieldName) + "', found '" +
                                 getShortSourceValue(valueEntry.getValue()) + "'");
                         returnVal = false;
                     }
                 } else if (!targetType.sealed) {
-                    if (getConvertibleTypesFromJson(valueEntry.getValue(), restFieldType, fieldNameLong,
-                            unresolvedValues, errors).size() != 1) {
+                    if (getConvertibleTypeFromJson(valueEntry.getValue(), restFieldType, fieldNameLong,
+                            unresolvedValues, errors, allowNumericConversion) ==  null) {
                         addErrorMessage(errors.size() - initialErrorCount, errors, "value of field '" +
```

Comment:
Should we add this error messages to Messagebundle.properties. May be we can do as separate task

Reply:
This is for creating a String which will only be a part of the error message. That will be then used with an error message already in  Messagesbundle.properties. So, we can also keep this like this.

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/37822 to do this task.